src/lib/table/table-data-source.ts

```diff
@@ -276,14 +280,24 @@ export class MatTableDataSource<T> extends DataSource<T> {
    */
   _updatePaginator(filteredDataLength: number) {
     Promise.resolve().then(() => {
-      if (!this.paginator) { return; }
+      const paginator = this.paginator;
```

Comment:
Why are we creating a new const for this?

Reply:
We were using `this.paginator` in 4-5 different places. I moved it out into a const because it minifies better.